airbyte/caches/base.py

```diff
@@ -146,6 +149,34 @@ def get_pandas_dataframe(
         engine = self.get_sql_engine()
         return pd.read_sql_table(table_name, engine, schema=self.schema_name)
 
+    def get_arrow_dataset(
```

Comment:
kept return type as Any because this is return type of ds.dataset()

Reply:
I'd prefer to be explicit if we can. Can we use `pd.dataset` (or `Dataset`) as the return type?

Reply:
<img width="843" alt="Screenshot 2024-06-29 at 10 00 52 AM" src="https://github.com/airbytehq/PyAirbyte/assets/100823015/b5f41e41-5729-4f37-bc92-2523e52ca20b">
Gone through official docs, return type can be Dataset

https://arrow.apache.org/docs/python/generated/pyarrow.dataset.dataset.html